src/NUnitFramework/tests/Internal/TestExecutionContextTests.cs

```diff
@@ -475,6 +475,25 @@ public async Task AsyncTestCanAccessOutWriter()
 
 #endregion
 
+#region SendMessage
+        [Test]
+        public void CanSendMessage()
+        {
+            TestExecutionContext.CurrentContext.SendMessage("destination", "message");
+
+#if !NETCOREAPP1_1
+            var listener = TestExecutionContext.CurrentContext.Listener as QueuingEventListener;
+            Assert.That(listener.Events.Count, Is.EqualTo(1));
```

Comment:
Should probably assert that the count is equal to one greater than the previous count?

Also, can you mark this test `[NonParallelizable]` to draw attention to the fact that this method cannot be safely run in parallel with other tests?

Reply:
It seems we cannot verify queue of the events.

Moreover it seems `SendMessage` method doesn't add TestEvent immediately into the queue. https://ci.appveyor.com/project/CharliePoole/nunit/build/3.11.0-ci-05907-pr-3018#L1001

Currently I removed all assertions which are related to the queue. Only one way I see that's we can dequeue TestEvents until we face our specific TestEvent, this is kind of awaiting. But this way may affect others tests which are working with listener's queue. Need your input here.